src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -172,14 +175,31 @@ private IEnumerable<string> GetViewNames()
             {
                 if (typeof(IEnumerable).IsAssignableFrom(fieldType))
                 {
+                    if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
```

Comment:
You repeat this logic twice for `IFormFile`/`IFormFileCollection`. In both cases they take precedence over  the interfaces/class types. Can you instead place them as their own else if statements before the `else if (fieldType.IsInterface())`? That way the logic is only ever done once and essentially does the same thing.